### ISSUE TITLE
var-naming[no-role-prefix] false positive on `cacheable` property of ansible.builtin.set_fact

### ISSUE THREAD

**Issue by infracaninophile** [ansible/ansible-lint#3483]
##### Summary

`ansible.builin.set_fact` can take a `cacheable` property to control longer term retention on facts.  `ansible-lint` doesn't
distinguish this from the name of a fact to be created, and this triggers the `var-naming[no-role-prefix]` violation improperly.

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```console (paste below)
ansible-lint --version
ansible-lint 6.16.2 using ansible-core:2.14.3 ruamel-yaml:0.17.26 ruamel-yaml-clib:0.2.7
```

- ansible installation method: OS package -- FreeBSD ports
- ansible-lint installation method: OS package -- FreeBSD ports

##### STEPS TO REPRODUCE

Use `ansible.builtin.set_fact` with `cacheable` in a role.  

Playbook: `foo.yaml`

```console (paste below)
---
- name: Illustrate ansible-lint false positive with set_fact and cacheable
  hosts: localhost
  roles:
    - foo_role
```

Role `foo_role/tasks/main.yaml`

```console
---

- name: Using set_fact with cacheable
  ansible.builtin.set_fact:
    foo_role_fact: bar
    cacheable: true
```

##### Desired Behavior

Possible security bugs should be reported via email to `security@ansible.com`

##### Actual Behavior

Please give some details of what is happening.
Include a [minimum complete verifiable example] with:

- minimized playbook to reproduce the error
- the output of running ansible-lint including the command line used

```console
% ansible-lint foo.yml
WARNING  Listing 1 violation(s) that are fatal
var-naming[no-role-prefix]: Variables names from within roles should use role_name_ as a prefix. (set_fact: cacheable)
roles/foo_role/tasks/main.yaml:5 Task/Handler: Using set_fact with cacheable

Read documentation for instructions on how to ignore specific rule violations.

                    Rule Violation Summary
 count tag                        profile rule associated tags
     1 var-naming[no-role-prefix] basic   idiom

Failed after min profile: 1 failure(s), 0 warning(s) on 2 files.
```

- if you're getting a stack trace, also the output of
  `ansible-playbook --syntax-check playbook`

```paste below
% ansible-playbook --syntax-check foo.yml

playbook: foo.yml
```

[minimum complete verifiable example]: http://stackoverflow.com/help/mcve
